RBQFetchedResultsController/Source/RBQFetchedResultsController.m

```diff
@@ -817,7 +817,14 @@ - (void)calculateChangesWithAddedSafeObjects:(NSSet *)addedSafeObjects
 
             [self runOnMainThread:^(){
                 [weakSelf.delegate controllerWillChangeContent:weakSelf];
+                if (useSem) {
```

Comment:
This method is way too long holy smokes.

So the problem is that the executing thread are doing things but it needs to be be blocked by main?  Looks like you found the solution lower in the code 👍